llvm/lib/Target/Mips/MipsISelLowering.cpp

```diff
@@ -876,21 +876,48 @@ static SDValue performANDCombine(SDNode *N, SelectionDAG &DAG,
 static SDValue performORCombine(SDNode *N, SelectionDAG &DAG,
                                 TargetLowering::DAGCombinerInfo &DCI,
                                 const MipsSubtarget &Subtarget) {
-  // Pattern match INS.
-  //  $dst = or (and $src1 , mask0), (and (shl $src, pos), mask1),
-  //  where mask1 = (2**size - 1) << pos, mask0 = ~mask1
-  //  => ins $dst, $src, size, pos, $src1
   if (DCI.isBeforeLegalizeOps() || !Subtarget.hasExtractInsert())
     return SDValue();
 
   SDValue And0 = N->getOperand(0), And1 = N->getOperand(1);
   unsigned SMPos0, SMSize0, SMPos1, SMSize1;
   ConstantSDNode *CN, *CN1;
+  uint64_t Pos = 0;
 
   // See if Op's first operand matches (and $src1 , mask0).
   if (And0.getOpcode() != ISD::AND)
     return SDValue();
 
+  if (And0.getOpcode() == ISD::AND && And1.getOpcode() == ISD::SHL) {
+    // Pattern match INS.
+    //   $dst = or (and $src1, (2**size0 - 1)), (shl $src2, mask1)
+    //   where size0 = mask1, pos + size = 32, SMPos0 = 0
```

Comment:
what is `pos`? what is `SMPos0`? They don't appear in the previous line.

Reply:
I delete this second line comment. `pos`  is in the next line `ins`, and `SMPos0` is in the next `isShiftedMask_64`.

Reply:
I accidentally deleted this branch and resubmitted a new PR https://github.com/llvm/llvm-project/pull/103017.